cutil/logger/logger.go

```diff
@@ -137,6 +125,20 @@ func Warning(format string, a ...interface{}) {
 	}
 }
 
+func extractLoggerArgs(format string, a ...interface{}) ([]interface{}, io.Writer) {
```

Comment:
@metmajer at this point I think refactoring the logger makes perfect sense.. it started off one way and and as the project grows it's turning into something else...

My proposal here would be the following:

1. Merge this *as is* to keep our velocity up
2. Open up issues to track the new changes to the logger
3. Update the logger over time

Also another thought would be - would it make sense to move the logger into it's own repository? We own the `kubicorn` org but aren't really using it... thoughts?